### PR TITLE
Update test framework

### PR DESCRIPTION
**Pull request by floryst**
### PR and Code Checklist

- [x] [semantic-release](https://github.com/semantic-release/semantic-release) commit messages
- [x] Run `npm run reformat` to have correctly formatted code

### Context
- #2104

### Changes

This results in quite some big changes, so do let me know if some choices and decisions do not make sense to you. Going commit-by-commit will make review much more straightforward.

I ended up dropping `karma-tap` because there's a race condition issue where some of the first tests to run do not get picked up by `karma-tap`, resulting in some tests just not being reported.

I'm keeping the percentage-based mismatch tolerance/threshold measure, since tracking down acceptable raw pixel count thresholds for each test is too much work.

- [x] overhauled test framework to report HTML test results
- [x] nicer image test output
- [x] globbed tests rather than Sources/tests.js
- [x] bad image tests have been skipped (see #2102, #2103, and #2104)
- [x] Documentation and TypeScript definitions were updated to match those changes

### Results
- `Sources/tests.js` is no longer something that needs to be maintained. Tests are picked up by glob.
- broken image tests are skipped (for now)
- Here is a sample of what the generated Test-Report.html looks like:

image

### Testing
- [x] This change adds or fixes unit tests
- [x] All tests complete without errors on the following environment:
  - **vtk.js**: latest (v21)

### CODE REVIEW

File: Sources/Widgets/Core/WidgetManager/test/testWidgetManager.js [hunk 1]
```diff
@@ -128,9 +128,10 @@ test.onlyIfWebGL('Test getPixelWorldHeightAtCoord', (t) => {
   }
 
   function testScaleInPixelsWithParallel() {
-    t.comment(
+    t.skip(
```
Comment (jourdain) [inline Sources/Widgets/Core/WidgetManager/test/testWidgetManager.js:131]: is the goal here is to print in output. Does .skip() do it?

**Review by jourdain — APPROVED** [review]
LGTM

**Comment by finetjul** [comment]
LGTM

File: Sources/Widgets/Core/WidgetManager/test/testWidgetManager.js (hunk 1 above)
Comment (finetjul) [reply Sources/Widgets/Core/WidgetManager/test/testWidgetManager.js:131]: I think @floryst skipped it because the tests is currently failing.

**Comment by martinken** [comment]
Let me quickly try this on windows and on WebGPU

**Review by martinken — APPROVED** [review]
Works on windows and WebGPU

WebGPU fails 6 tests, some look like real failures and some like threshold issues but that seems reasonable for now. I'll clean up WebGPU on a different MR eventually.

LGTM +1

**Comment by finetjul** [comment]
@floryst it might be worth updating [CONTRIBUTING.rst](https://github.com/Kitware/vtk-js/blob/master/CONTRIBUTING.md). It contains old test framework advice.

**Comment by floryst** [comment]
Good point, I'll make a note of that.

**Pull request merged**